docs/cpp/customize-cpp-settings.md

```diff
@@ -0,0 +1,175 @@
+
+- `intelliSenseMode`
+  The IntelliSense mode to use that maps to an architecture-specific variant of MSVC, gcc, or Clang. If not set or if set to `${default}`, the extension chooses the default for that platform.
+
+  Platform defaults:
+  - Windows: `windows-msvc-x64`
+  - Linux: `linux-gcc-x64`
+  - macOS: `macos-clang-x64`
+
+  IntelliSense modes that only specify `<compiler>-<architecture>` variants (for example, `gcc-x64`) are legacy modes and are automatically converted to the `<platform>-<compiler>-<architecture>` variants based on the host platform.
+
+- `includePath`
+  An include path is a folder that contains header files (such as `#include "myHeaderFile.h"`) that are included by a source file. Specify a list of paths for the IntelliSense engine to use while searching for included header files. Searching on these paths is not recursive. Specify `/**` at the end of the path to indicate recursive search. For example, `${workspaceFolder}/**` searches through all subdirectories while `${workspaceFolder}` will not. If you are on Windows with Visual Studio installed, or if a compiler is specified in the `compilerPath` setting, the system include paths should not be listed here.
+
+- `defines`
+  A list of preprocessor definitions for the IntelliSense engine to use while parsing files. Optionally, use `=` to set a value, for example `VERSION=1`.
+
+- `cStandard`
+  The version of the C language standard to use for IntelliSense. For example, `c17`, `gnu23`, or `${default}`. Note: GNU standards are only used to query the set compiler to get GNU defines, and IntelliSense emulates the equivalent C standard version.
+
+- `cppStandard`
+  The version of the C++ language standard to use for IntelliSense. For example, `c++20`, `gnu++23`, or `${default}`. Note: GNU standards are only used to query the set compiler to get GNU defines, and IntelliSense emulates the equivalent C++ standard version.
+
+- `configurationProvider`
+  The ID of a VS Code extension that can provide IntelliSense configuration information for source files. For example, use the VS Code extension ID `ms-vscode.cmake-tools` to provide configuration information from the CMake Tools extension. If you specified a configurationProvider, the configuration it provides takes precedence over your other settings in `c_cpp_properties.json`.
+
+  A `configurationProvider` candidate extension must implement [vscode-cpptools-api](https://github.com/microsoft/vscode-cpptools-api).
+
+- `mergeConfigurations`
+  Set to `true` to merge include paths, defines, and forced includes with those from a configuration provider.
+
+- `windowsSdkVersion`
+  The versions of the Windows SDK include path to use on Windows, for example `10.0.17134.0`.
+
+- `macFrameworkPath`
+  A list of paths for the IntelliSense engine to use while searching for included headers from Mac frameworks. Only supported on configurations for macOS frameworks.
```

Comment:
I'm not understanding what `Only supported on configurations for macOS frameworks.` is supposed to mean.

Reply:
Removed - we support this on all OS now